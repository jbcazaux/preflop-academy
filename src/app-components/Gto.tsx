import gto from 'data/gto'
import Action from 'domain/action'
import Hand from 'domain/hand'
import Position from 'domain/position'

import ActionComponent from 'components/Action'

interface Props {
  hero: Position
  actions: ReadonlyArray<Action>
  hand: Hand
}

const Gto = ({ hero, actions, hand }: Props) => {
  const move = gto(
    hero,
    actions.map(a => a.position),
    hand
  )
  return <ActionComponent>You should : {move || 'N/A'}</ActionComponent>
}

export default Gto
