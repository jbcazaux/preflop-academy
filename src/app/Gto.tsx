import Hand from 'domain/hand'
import gto from 'data/gto'
import Position from 'domain/position'
import ActionComponent from 'components/Action'
import Action from 'domain/action'

interface Props {
  hero: Position
  actions: ReadonlyArray<Action>
  hand: Hand
}

const Gto: React.FC<Props> = ({ hero, actions, hand }) => {
  const move = gto(
    hero,
    actions.map(a => a.position),
    hand
  )
  return <ActionComponent>You should : {move || 'N/A'}</ActionComponent>
}

export default Gto
