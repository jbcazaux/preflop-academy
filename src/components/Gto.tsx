import Hand from 'domain/hand'
import React from 'react'
import gto from '../data/gto'
import styled from 'styled-components'

const Advice = styled.div`
  display: flex;
  margin: 10px auto;
  font-weight: bold;
  font-size: 2em;
`

interface Props {
  buttonPostion: number
  raisePositions: ReadonlyArray<number>
  hand: Hand
}

const Gto: React.FC<Props> = ({ buttonPostion, raisePositions, hand }) => {
  const advice = gto(buttonPostion, raisePositions, hand)
  return <Advice>{advice}</Advice>
}

export default Gto
