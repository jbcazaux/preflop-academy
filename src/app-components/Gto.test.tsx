import { render } from '@testing-library/react'
import Hand from 'domain/hand'
import Move from 'domain/move'
import Position from 'domain/position'

import 'jest-styled-components'

import Gto from './Gto'

jest.mock('data/gto', () => () => Move.OPEN)
describe('Gto', () => {
  it('displays whatever gto says', () => {
    const { container } = render(<Gto hero={Position.B} actions={[]} hand={Hand.newHand} />)

    expect(container).toMatchSnapshot()
  })
})
