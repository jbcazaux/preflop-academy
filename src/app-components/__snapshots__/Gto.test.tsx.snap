// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gto displays whatever gto says 1`] = `
<div>
  <div
    class="action"
  >
    You should : 
    OPEN
  </div>
</div>
`;
